lib/hammer_cli/options/option_family.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module HammerCLI
+  module Options
+    class OptionFamily
+      attr_reader :children, :description
+
+      def initialize(options = {})
+        @all = []
+        @children = []
+        @options = options
+        @creator = options[:creator] || Class.new(HammerCLI::Apipie::Command)
+        @prefix = options[:prefix]
+        @description = options[:description]
+        @root = options[:root] || options[:aliased_resource] || options[:referenced_resource]
+      end
+
+      def switch
+        return if @parent.nil? && @children.empty?
+        return @parent.help_lhs if @children.empty?
+
+        root = @root || @parent.aliased_resource || @parent.referenced_resource
+
+        main_switch = optionamize("--#{@prefix}#{root}")
+        main_switch = main_switch.each_char
+                                 .zip(*all.map(&:switches).flatten.map(&:each_char))
+                                 .select { |a, b| a == b }.transpose.first.join
+        suffixes = all.map do |m|
+          m.switches.map { |s| s.gsub(main_switch, '') }
+        end.flatten.reject(&:empty?).sort { |x, y| x.size <=> y.size }
+        "#{main_switch}[#{suffixes.join('|')}]"
+      end
+
+      def all
+        @children + [@parent].compact
+      end
+
+      def head
+        @parent
+      end
+
+      def parent(switches, type, description, opts = {}, &block)
+        raise StandardError, 'Option family can have only one parent' if @parent
+
+        @parent = new_member(switches, type, description, opts, &block)
+      end
+
+      def child(switches, type, description, opts = {}, &block)
+        child = new_member(switches, type, description, opts, &block)
+        @children << child
+        child
+      end
+
+      def adopt(child)
+        child.family = self
+        @children << child
+      end
+
+      private
+
+      def new_member(switches, type, description, opts = {}, &block)
+        opts = opts.merge(@options)
+        opts[:family] = self
+        if opts[:deprecated]
+          handles = [switches].flatten
+          opts[:deprecated] = opts[:deprecated].select do |switch, _msg|
+            handles.include?(switch)
+          end
+        end
+        @creator.instance_eval do
+          option(switches, type, description, opts, &block)
+        end
+      end
+
+      def optionamize(name_candidate)
```

Comment:
While I do get why the method is named like this, the name just looks strange to me. Can you think of a better name? I can share my suggestions if you'd like...

Reply:
@xprazak2, actually we already have that method in option builders https://github.com/theforeman/hammer-cli/blob/16a30c6e93cecbbf5321f402b1abb75ca428a8b4/lib/hammer_cli/option_builder.rb#L15

But I'm open to a new better name.

Reply:
Oh, I thought it was a new one. In that case, I think it is ok to keep.